Tests/SpeechToTextV1Tests/SpeechToTextTests.swift

```diff
@@ -73,6 +89,196 @@ class SpeechToTextTests: XCTestCase {
+        let failToCreateCustomization =  { (error: Error) in
+            XCTFail("Failed to create the new customization: \(error)")
+            return
+        }
+        
+        let description = "Create a new customization."
+        let expectation = self.expectation(description: description)
+        speechToText.createCustomization(
+            withName: self.trainedCustomizationName,
+            withBaseModelName: "en-US_BroadbandModel",
+            description: self.trainedCustomizationDescription,
+            failure: failToCreateCustomization) { customization in
+                
+            self.trainedCustomizationID = customization.customizationID
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    func addCorpusToTrainedCustomization() {
+        
+        let failToCreateCorpus =  { (error: Error) in
+            XCTFail("Failed to create the new corpus: \(error)")
+            return
+        }
+        
+        guard let corpusFile = loadFile(name: "healthcare", withExtension: "txt") else {
+            XCTFail("Failed to load file needed to create the corpus.")
+            return
+        }
+        
+        let description = "Add a corpus for the customization."
+        let expectation = self.expectation(description: description)
+        speechToText.addCorpus(
+            withName: corpusName,
+            fromFile: corpusFile,
+            customizationID: self.trainedCustomizationID,
+            failure: failToCreateCorpus) {
+                
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    func trainCustomizationWithCorpus() {
+        
+        let failToGetCorpusStatus = { (error: Error) in
+            XCTFail("Failed to get the status of the new corpus: \(error)")
+            return
+        }
+        
+        let failToTrainCustomization = { (error: Error) in
+            XCTFail("Failed to train the new customization: \(error)")
+            return
+        }
+        
+        let failToGetCustomizationStatus = { (error: Error) in
+            XCTFail("Failed to get the status of the new customization: \(error)")
+            return
+        }
+        
+        var trained = false
+        var tries = 0
+        while(!trained) {
+            tries += 1
+            let description = "Wait until the corpus is processed before training the customization."
+            let expectation = self.expectation(description: description)
+            
+            speechToText.getCorpus(
+                withName: corpusName,
+                customizationID: self.trainedCustomizationID,
+                failure: failToGetCorpusStatus) { corpus in
+                    
+                    if corpus.status == .analyzed {
+                        trained = true
+                    }
+                    expectation.fulfill()
+            }
+            waitForExpectations()
+            
+            if tries > 10 {
```

Comment:
Ahh this is why code reviews are great haha.

I just realized: I mis-named my variables here. In this case, you're right; it should be 
```
repeat {
   ...
} while(!processed)
```
That is much, *much* clearer than what I currently have.

However, we'd still need the `if tries > 10` check. Tests only timeout if the `expectation.fulfill()` isn't called before 15 seconds, or however long we set the timeout to be. It doesn't take into account how long that test as a whole has run for. 

The issue here is that the `getCorpus` call is successful, but we don't exit the loop until `corpus.status == .analyzed`, which might never happen if there's an error. If there's an error, `corpus.status == .undetermined`. 

Do you think there's a better method of doing this? The only other thing I can think of is to   fail immediately if `corpus.status == .undetermined`, which might be better than continuously running the loop for 10 times. So we either exit after 10 tries (it's taking too long to train, try again later) or fail the test if the corpus errors (status == .undetermined).